storage.go

```diff
@@ -2,35 +2,283 @@ package azfile
 
 import (
 	"context"
+	"encoding/base64"
 	"io"
+	"strconv"
 
+	"github.com/Azure/azure-storage-file-go/azfile"
+
+	"github.com/beyondstorage/go-storage/v4/pkg/iowrap"
+	"github.com/beyondstorage/go-storage/v4/services"
 	. "github.com/beyondstorage/go-storage/v4/types"
 )
 
 func (s *Storage) create(path string, opt pairStorageCreate) (o *Object) {
-	panic("not implemented")
+	rp := s.getAbsPath(path)
+
+	if opt.HasObjectMode && opt.ObjectMode.IsDir() {
+		o = s.newObject(true)
+		o.Mode |= ModeDir
+	} else {
+		o = s.newObject(false)
+		o.Mode |= ModeRead
+	}
+
+	o.ID = rp
+	o.Path = path
+
+	return o
+}
+
+func (s *Storage) createDir(ctx context.Context, path string, opt pairStorageCreateDir) (o *Object, err error) {
+	rp := s.getAbsPath(path)
+
+	attribute := azfile.FileAttributeNone
+
+	properties := azfile.SMBProperties{
+		FileAttributes: &attribute,
+	}
+
+	_, err = s.client.NewDirectoryURL(path).Create(ctx, nil, properties)
```

Comment:
Create directory doesn't support `overwrite`, so we need to check whether the dir exists here.
ref: https://docs.microsoft.com/en-us/rest/api/storageservices/create-directory#remarks